src/celeritas/optical/detail/OpticalLaunchAction.cc

```diff
@@ -107,19 +113,18 @@ std::string_view OpticalLaunchAction::description() const
 /*!
  * Build state data for a stream.
  */
-auto OpticalLaunchAction::create_state(MemSpace m,
-                                       StreamId sid,
-                                       size_type size) const -> UPState
+auto OpticalLaunchAction::create_state(MemSpace m, StreamId sid, size_type) const
+    -> UPState
 {
     if (m == MemSpace::host)
     {
         return std::make_unique<optical::CoreState<MemSpace::host>>(
-            *optical_params_, sid, size);
+            *optical_params_, sid, num_track_slots_);
```

Comment:
This may not be the best way to set the state size... another option might be to store the size in each of the aux params classes (although this should be the only case where the size is different?).